codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/c2j/query/service-2.json

```diff
@@ -59,6 +59,17 @@
       },
       "authtype": "none"
     },
+    "OperationWithRequestCompression": {
+      "name": "APostOperation",
+      "http": {
+        "method": "POST",
+        "requestUri": "/"
+      },
+      "authtype": "none",
+      "requestCompression": {
+        "encodings": ["gzip"]
+      }
+    },
```

Comment:
Thanks for adding the test. I think we should add tests for other protocols as well since it seems the codegen code for different protocols is in different files.